base/pair.jl

```diff
@@ -28,18 +28,23 @@ foo
 7
 ```
 """
-struct Pair{A,B}
+struct Pair{A, B}
     first::A
     second::B
+    function Pair{A, B}(@nospecialize(a), @nospecialize(b)) where {A, B}
```

Comment:
I guess this is fine for now but we might want to use `@nospecialize` as a hint to inference as well at some point.

Reply:
Possibly. Right now though, I think it would be a very bad hint, so I'm not sure it would be a good idea.

Reply:
Maybe add a comment explaining why this constructor is needed? It's very intriguing to me.

Reply:
Basically, this expresses the idea that if you have a small function without any loops, and inference failed to inline it, that probably means the callsite is actually very dynamic – and you therefore actually want to do the minimal compilation work